code/iaas/lb-instance/src/main/java/io/cattle/platform/lb/instance/dao/impl/LoadBalancerInstanceDaoImpl.java

```diff
@@ -0,0 +1,51 @@
+package io.cattle.platform.lb.instance.dao.impl;
+
+import static io.cattle.platform.core.model.tables.NetworkTable.NETWORK;
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.model.LoadBalancer;
+import io.cattle.platform.core.model.LoadBalancerHostMap;
+import io.cattle.platform.core.model.Network;
+import io.cattle.platform.db.jooq.dao.impl.AbstractJooqDao;
+import io.cattle.platform.lb.instance.dao.LoadBalancerInstanceDao;
+import io.cattle.platform.object.ObjectManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import com.netflix.config.DynamicStringProperty;
+
+public class LoadBalancerInstanceDaoImpl extends AbstractJooqDao implements LoadBalancerInstanceDao {
+
+    private static final DynamicStringProperty MANAGED_NETWORK_NAME = ArchaiusUtil
+            .getString("lb.instance.network.name");
+
+    GenericMapDao mapDao;
+    ObjectManager objectManager;
+
+    @Override
+    public List<Long> getLoadBalancerHosts(long lbId) {
+        List<Long> hostIds = new ArrayList<Long>();
+        for (LoadBalancerHostMap map : mapDao.findToRemove(LoadBalancerHostMap.class, LoadBalancer.class, lbId)) {
+            hostIds.add(map.getHostId());
+        }
+        return hostIds;
+    }
+
+    @Inject
+    public void setMapDao(GenericMapDao mapDao) {
+        this.mapDao = mapDao;
+    }
+
+    @Inject
+    public void setObjectManager(ObjectManager objectManager) {
+        this.objectManager = objectManager;
+    }
+
+    @Override
+    public Network getManagedNetwork() {
+        return objectManager.findOne(Network.class, NETWORK.NAME, MANAGED_NETWORK_NAME.get());
```

Comment:
@ibuildthecloud Darren, if you decide to go with putting a networkId on the LB, do you think we should allow only Targets (containers) from the same network to be added to this LB?